From c69180696dccefed5da097d624bea3964ed17a47 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 11 Apr 2016 15:20:49 +0300
Subject: [PATCH 0164/1345] fix: mvpp2x: Removed mac reset in RGMII mode

commit  bd5211d6272a7eb57037bee6540a8380675155c3 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- fix Tx stuck issue in RGMII mode
- MAC reset cause TX stuck in RGMII mode, by default
  MAC in reset and reset function removed from gop
  port init

Change-Id: Ia21aabbb8290c95323b2012b5764f943ffa9e512
Reviewed-on: http://vgitil04.il.marvell.com:8080/28928
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c |    2 +-
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    6 ------
 2 files changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
index 6cab141..5443e2b 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
@@ -880,7 +880,7 @@ int mv_gop110_port_init(struct gop_hw *gop, struct mv_mac_data *mac)
 
 	switch (mac->phy_mode) {
 	case PHY_INTERFACE_MODE_RGMII:
-		mv_gop110_gmac_reset(gop, mac_num, RESET);
+		/* As workaround do not reset mac before MAC configuration */
 		/* configure PCS */
 		mv_gop110_gpcs_mode_cfg(gop, mac_num, false);
 
diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index f43eda5..028d911 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -2910,12 +2910,6 @@ int mv_pp2x_open(struct net_device *dev)
 #endif
 #if !defined(CONFIG_MV_PP2_FPGA) && !defined(CONFIG_MV_PP2_PALLADIUM)
 	/* Port is init in uboot */
-#if !defined(OLD_UBOOT)
-	if (port->mac_data.phy_mode == PHY_INTERFACE_MODE_RGMII ||
-	    port->mac_data.phy_mode == PHY_INTERFACE_MODE_KR ||
-	    port->mac_data.phy_mode == PHY_INTERFACE_MODE_SGMII)
-		port->mac_data.flags |= MV_EMAC_F_INIT;
-#endif
 
 	if (port->priv->pp2_version == PPV22)
 		mvcpn110_mac_hw_init(port);
-- 
1.7.9.5

